src/auth/auth.service.ts

```diff
@@ -25,47 +22,84 @@ export class AuthService {
+      throw new HttpException(
+        "Token coundn't be created in the database",
+        HttpStatus.INTERNAL_SERVER_ERROR,
+      );
     }
-    this.configCollection[serverId].knownClients = [
-      ...this.configCollection[serverId].knownClients,
-      clientId,
-    ];
+
     //TODO token-expiry
     const signedToken = this.jwtService.sign(payload, { expiresIn: '1y' });
     const decoded: any = this.jwtService.decode(signedToken);
     const expiresIn: number = decoded.exp - Math.round(Date.now() / 1000);
-    return { ...payload, accessToken: signedToken, expiresIn, tokenType };
+    return { ...payload, accessToken: signedToken, expiresIn };
   }
 
-  public validateClient(payload: TokenPayload): boolean {
+  public async validateClient(payload: TokenPayload): Promise<boolean> {
     const { keyspace, clientId } = payload;
-    if (
-      this.configCollection[keyspace] &&
-      this.configCollection[keyspace].knownClients.includes(clientId)
-    ) {
-      return true;
+    let token: string = null;
+    try {
+      token = await this.astraService
+        .get<string>(clientId, keyspace, 'tokens')
+        .toPromise();
+    } catch {
+      return false;
+    }
+    if (!token) {
+      return false;
     }
-    return false;
+    return true;
   }
 
-  public removeClient(token: string) {
+  public async removeClient(token: string) {
+    let deleteSuccess = false;
+    let clientId: string;
+    let keyspace: string;
+
     if (!token)
       throw new HttpException('Please provide token', HttpStatus.BAD_REQUEST);
+    try {
+      ({ clientId, keyspace } = this.jwtService.verify<TokenPayload>(token));
+    } catch (error) {
+      throw new HttpException(
+        "Token couldn't be verified",
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+    try {
+      await this.astraService.delete(clientId, keyspace, 'tokens').toPromise();
+      deleteSuccess = true;
+    } catch (error) {
+      throw new Error("Token couldn't be deleted");
+    }
+
+    return deleteSuccess;
+  }
 
-    const decoded = this.jwtService.decode(token) as TokenPayload;
+  public async getClientIds(keyspace: string) {
+    let clients;
+    try {
+      clients = await this.astraService.find(keyspace, 'tokens').toPromise();
+    } catch (error) {
+      throw new HttpException(
+        'Clients couldnt be retrieved',
+        HttpStatus.INTERNAL_SERVER_ERROR,
+      );
+    }
+    return { clients: Object.keys(clients) };
+  }
 
+  public validateToken(token: string, response: Response) {
     try {
-      this.configCollection[
-        decoded.keyspace
-      ].knownClients = this.configCollection[
-        decoded.keyspace
-      ].knownClients.filter((client) => client !== decoded.clientId);
-      console.log(this.configCollection);
-      return;
-    } catch (e) {
-      throw new HttpException('Invalid client id', HttpStatus.BAD_REQUEST);
+      this.jwtService.verify(token);
+      response.status(HttpStatus.OK).json({ valid: true });
```

Comment:
For now I'm just validating if the token is valid. 

-  Do we have to do more checks here?
- How should the response look like at this endpoint? Is it just returning `{valid: true/false}` Or does it return more information?
- Does this endpoint needs to be secured, or can anybody post data to it for checking if it is valid?

Reply:
Added the token-guard for using only with the bot to be consistent with all Auth-Endpoints

Reply:
Ok sure 👍

Reply:
> `response.status(HttpStatus.OK)`

I think it might make more sense to have the response code decision made in the controller and the service to just return the `boolean`

Reply:
done